src/backend.py

```diff
@@ -0,0 +1,726 @@
+        return json_dumps({"in_ap_mode": False}), 200
+
+#
+# AP mode routes
+#
+def add_ap_mode_routes():
+    '''Routes only available in AP mode'''
+
+    @add_route("/api/in_ap_mode")
+    def app_inAPMode(request):
+        return json_dumps({"in_ap_mode": True}), 200
+
+    @add_route("/api/settings/set_vector_config", method="POST")
+    def app_setWifi(request):
+        '''Set the wifi SSID and password'''
+        from GameDefsLoad import list_game_configs
+        all_game_configs = list_game_configs().keys()
+        
+        data = request.data
+        if data['game_config_filename'] not in all_game_configs:
+            return f"Invalid game config filename {data['game_config_filename']}", 400
+        
+
+        ds_write_record("configuration",
+            {
+                "ssid": data['ssid'],
+                "password": data['wifi_password'],
+                "Gpassword": data['vector_password'],
+                "gamename": data['game_config_filename'],
+            }
+        )
+
+        Pico_Led.off()
+
+def connect_to_wifi():
+    from phew import connect_to_wifi as phew_connect, is_connected_to_wifi as phew_is_connected
+    wifi_credentials = ds_read_record("configuration", 0)
+    ssid = wifi_credentials["ssid"]
+    password = wifi_credentials["password"]
+
+    if not ssid:
+        return False # Wifi not configured
+    
+    for i in range(WIFI_MAX_ATTEMPTS):
+        print(f"Attempt {i+1} to connect to wifi ssid:{ssid}")
+        ip_address = phew_connect(ssid, password, timeout_seconds=10) 
+        if phew_is_connected():
+            print(f"Connected to wifi with IP address {ip_address}")
+            writeIP(ip_address)
+            Pico_Led.on()
+            from displayMessage import init
+            init(ip_address)
+            return True
+    
+    # check that we get signal form the ssid
+    import scanwifi
+    networks = scanwifi.scan_wifi2()
+    for network in networks:
+        if network['ssid'] == ssid:
+            faults.raise_fault(faults.WIFI01, f"Invalid wifi credentials for ssid: {ssid}")
+            return False
+    
+    # We have credentials but no signal so raise a fault
+    faults.raise_fault(faults.WIFI02, f"No wifi signal for ssid: {ssid}")
+    return False
+
+def go(ap_mode):
+    '''Start the server and run the main loop'''
+    
+    #TODO this was earlier in the code, might need moved to the top of this file OG comments:
+    #Allocate PICO led early - this grabs DMA0&1 and PIO1_SM0 before memory interfaces setup
+    #wifi uses PICO LED to indicate status (since it is on wifi chip via spi also)   
+    Pico_Led.off()
+    gc_threshold(2048 * 6) 
+
+    if not ap_mode:
+        # try to connect to WiFi
+        try:
+            if not connect_to_wifi():
+                ap_mode = True # If we can't connect to wifi, start in AP mode
+        except Exception as e:
+            ap_mode = True
+            faults.raise_fault(faults.WIFI00, e)
```

Comment:
They way I've written this means if we fail to connect we go to AP mode.

Reply:
Feature to try wifi every x seconds incase it is only down temporarily

Reply:
implemnted